sagemaker_neo_compilation_jobs/deploy_pytorch_model_on_Inf1_instance/pytorch_torchvision_neo_on_Inf1.ipynb

```diff
@@ -0,0 +1,300 @@
+    "\n",
+    "Now that we have the compiled model, we will deploy it on an Amazon SageMaker endpoint. Inf1 instances in Amazon SageMaker are available in four sizes: ml.inf1.xlarge, ml.inf1.2xlarge, ml.inf1.6xlarge, and ml.inf1.24xlarge. In this example, we are using ``'ml.inf1.xlarge'`` for deploying our model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "predictor = neo_model.deploy(instance_type='ml.inf1.xlarge', initial_instance_count=1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "predictor.endpoint_name"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Invoking the endpoint\n",
+    "\n",
+    "Once the endpoint is ready, you can send requests to it and receive inference results in real-time with low latency. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's try to send a cat picture.\n",
+    "\n",
+    "![title](cat.jpg)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "import numpy as np\n",
+    "\n",
+    "sm_runtime = boto3.Session().client('sagemaker-runtime')\n",
+    "\n",
+    "with open('cat.jpg', 'rb') as f:\n",
+    "    payload = f.read()\n",
+    "\n",
+    "response = sm_runtime.invoke_endpoint(EndpointName=predictor.endpoint_name,\n",
+    "                                      ContentType='application/x-image',\n",
+    "                                      Body=payload)\n",
+    "print(response)\n",
+    "result = json.loads(response['Body'].read().decode())\n",
+    "print('Most likely class: {}'.format(np.argmax(result)))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Load names for ImageNet classes\n",
+    "object_categories = {}\n",
+    "with open(\"imagenet1000_clsidx_to_labels.txt\", \"r\") as f:\n",
+    "    for line in f:\n",
+    "        key, val = line.strip().split(':')\n",
+    "        object_categories[key] = val\n",
+    "print(\"Result: label - \" + object_categories[str(np.argmax(result))]+ \" probability - \" + str(np.amax(result)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Delete the Endpoint\n",
+    "Having an endpoint running will incur some costs. Therefore as a clean-up job, we should delete the endpoint."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sess.delete_endpoint(predictor.endpoint_name)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Revert SageMaker version\n",
+    "If needed, uninstall the SageMaker version 2.10.0"
```

Comment:
Provide code to do this for them.

Reply:
I plan to change install from version==2.10.0 to version>=2.10.0. If that is the case, maybe we don't need to revert the version?

Reply:
Reverting to the original SDK version is so that we don't mess up customer's environment. 

So in the beginning, you'd have to get the SDK they running, save it, install version you need for this notebook, then at the end revert back to whatever they had.

Reply:
[This might require a notebook restart as well]

Reply:
I have added the revert version, could you help to take a look?

Reply:
Close! I think the CLI is grumpy about having a str var in it?

`!{sys.executable} -m pip install -qU sagemaker==sagemaker_version`

I get "version not found" followed by a list of SM SDK versions

Reply:
Ah, I was not able to find the solution at first place. Now I will update that with working command.